server.go

```diff
@@ -367,7 +371,6 @@ func (srv *Server) Shutdown() error {
 		return &Error{err: "server not started"}
 	}
 	srv.started = false
-	srv.lock.Unlock()
```

Comment:
Ah, I see you merged this in. I was totally willing to do this one but was going to mention being a little worried about future accidents because the scope of data the lock would cover would become more broad.

But, hey, cool with me either way.